beesdoo_purchase/models/purchase.py

```diff
@@ -39,3 +39,62 @@ def write(self, vals):
                     partner_ids=[new_partner], subtype_ids=[]
                 )
         return super(PurchaseOrder, self).write(vals)
+
+    @api.multi
+    def action_toggle_adapt_purchase_price(self):
+        for order in self:
+            for line in order.order_line:
+                line.adapt_purchase_price ^= True
```

Comment:
Hum, nice use of a XOR. :p But, personally, I prefer (if I did'nt miss something):
```
line.adapt_purchase_price = (
    not line.adapt_purchase_price
)
```
Find this more readable. I'm interested in your opinion. :)

As it is a matter of preference, your choice is OK for my review.

Reply:
I agree, your suggestion is more readable. Since this new feature needs to be improved, I'll keep this in mind to change it in a future PR :)